tutorial_code/embed_loops.py

```diff
@@ -13,57 +13,132 @@
-    if try_to_load:
+    sublattice_size = kwargs.pop(
+        "sublattice_size",
+        min(lattice_size_lower_bound(S=G, T=A) + 1, lattice_size_upper_bound(T=A)),
+    )
+
+    if not isinstance(sublattice_size, int) or sublattice_size <= 0:
+        raise ValueError(
+            f"'sublattice_size' must be a positive integer. Received {sublattice_size}."
+        )
+
+    print(
+        "Creating embeddings may take several minutes."
+        "\nTo accelerate the process a smaller lattice (L) might be "
+        "considered and/or the search restricted to max_num_emb=1."
+    )
+    max_num_emb = kwargs.pop("max_num_emb", float("Inf"))
+    embedder_kwargs = {"timeout": kwargs.pop("timeout", 10)}
+    embeddings = embeddings_to_array(
+        find_sublattice_embeddings(
+            S=G,
+            T=A,
+            sublattice_size=sublattice_size,
+            max_num_emb=max_num_emb,
+            embedder_kwargs=embedder_kwargs,
+            **kwargs,
+        ),
+        node_order=sorted(G.nodes()),
+        as_ndarray=True,
+    )
+
+    if embeddings.size == 0:
+        raise ValueError(
+            "No feasible embeddings found. "
+            "\nModifying the source (lattice) and target "
+            "(processor), or find_sublattice_embeddings arguments "
+            "such as timeout may resolve the issue."
+        )
+
+    if use_cache:
         try:
-            embeddings = np.loadtxt(cache_filename, dtype=int)
-            print(f'Loaded embedding from file {cache_filename}')
-            return embeddings
-        except (ValueError, FileNotFoundError) as e:
-            print(f"Failed to load {cache_filename} with `np.loadtxt`")
-            print("Error:", e)
-            print("Finding embedding via raster embedding search instead.")
-    embeddings = raster_embedding_search(A, G, raster_breadth=raster_breadth)
-
-    os.makedirs('cached_embeddings/', exist_ok=True)
-    np.savetxt(cache_filename, embeddings, fmt='%d')
+            os.makedirs("cached_embeddings/", exist_ok=True)
+            np.savetxt(cache_filename, embeddings, fmt="%d")
+        except:
+            warnings.warn("Embedding cache files could not be created")
 
     return embeddings
 
 
-def main():
-    L = 8  # Length of chain to embed
-    embeddings = embed_loops(L, raster_breadth=2)
+def main(max_num_emb=None):
+    from time import perf_counter
+
+    if max_num_emb is not None:
+        max_num_emb = max_num_emb
+
+    L = 2048  # L=2048 anticipate ~ 2.5 seconds on i7
+    sampler = MockDWaveSampler(topology_type="pegasus", topology_shape=[16])
+    t0 = perf_counter()
+    embeddings = embed_loops(sampler=sampler, L=L, max_num_emb=1, use_cache=False)
+    t1 = perf_counter() - t0
+    if embeddings.size >= 1:
+        print(f"Loop {L} embedding successfully found in {t1} seconds")
+    else:
+        print(f"Something is wrong, {L}x{L} embedding not found")
 
 
 if __name__ == "__main__":
-    main()
+    main(max_num_emb=None)
```

Comment:
None is the default, just leave as () for tidiness sake.

Reply:
I also modified for embed_lattice_square to make it same format.